src/Pact/Types/ChainId.hs

```diff
@@ -0,0 +1,46 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE DerivingStrategies #-}
+-- |
+-- Module      :  Pact.Types.Runtime
+-- Copyright   :  (C) 2019 Stuart Popejoy
+-- License     :  BSD-style (see the file LICENSE)
+-- Maintainer  :  Stuart Popejoy <stuart@kadena.io>,
+--                Emily Pillmore <emily@kadena.io>
+--
+-- ChainId data and its associated combinators
+--
+module Pact.Types.ChainId
+( -- * types
+  ChainId(..)
+  -- * optics
+, chainId
+) where
+
+import GHC.Generics
+
+import Control.DeepSeq
+import Control.Lens
+
+import Data.Aeson (ToJSON, FromJSON)
+import Data.Serialize (Serialize)
+import Data.String (IsString)
+import Data.Text
+
+import Pact.Types.Term (ToTerm(..))
+
+-- | Expresses unique platform-specific chain identifier.
+--
+newtype ChainId = ChainId { _chainId :: Text }
+  deriving (Eq, Show, Generic, IsString, ToJSON, FromJSON, Serialize)
+  deriving newtype (NFData)
+
+instance ToTerm ChainId where toTerm (ChainId i) = toTerm i
+
+instance Wrapped ChainId
+
+-- | Chain Id getter and setter which is a strict get/set pair
+--
+chainId :: Lens' ChainId Text
+chainId = lens _chainId (\_ t -> ChainId t)
```

Comment:
This was added to relieve a cycle in the import chain: 

```
ChainMeta --> Continuation --> Persistence --> Gas --> ChainMeta
```

The only reason `Continuation` was involved was because `ChainId` was a common import between `ChainMeta` and `Continuation`, with the `Gas <--> Persistence` cycle being the main culprit. We should figure out how to factor those two out of a cycle so that this file can go.